### ISSUE TITLE
Allow naming file with single extension method after the target of the extension

### ISSUE THREAD

**Issue by dalewking** [pinterest/ktlint#1521]
When a file contains a single top-level extension function should allow naming the file after the fully qualified function, just like is allowed for extension properties

## Steps to Reproduce
Given a class like this:
```
    class Foo {
        enum Status { A, B }
    }
```

If I create a file that only has a single extension method on Foo.Status like this:
```
    fun Foo.Status.image() {}
```

## Observed Behavior
Ktlint demands that the file be named Image.kt which is not descriptive and may even conflict if I had multiple similar functions in the same directory

## Expected Behavior
I should be able to name the file based on the target of the exension, so in this case FooStatusImage.kt.

Note that this is the behavior when it is an extension property, but not when it is an extension function.

## Your Environment
ktlint 0.46.1

**Comment by paul-dingemans** [comment]
I like the suggestion for the expected behavior. So either the name of the function `Image.kt` or `FooStatusImage.kt` should be acceptable names.

For now, you can work around this by suppressing the rule for this file specifically:
```
@file:Suppress("ktlint:filename")

fun Foo.Status.image() {}
```

**Comment by dalewking** [comment]
In my particular case I worked around it by converting it from a function to a property since it probably should have been a property all along

**Comment by PaulWoitaschek** [comment]
> In my particular case I worked around it by converting it from a function to a property since it probably should have been a property all along

That works sometimes but in general it can't be a solution.

In my case, the rule wants to name my file `Plus.kt`

```kotlin
operator fun PaddingValues.plus(other: PaddingValues): PaddingValues {
  val self = this
  return object : PaddingValues {

    override fun calculateLeftPadding(layoutDirection: LayoutDirection): Dp {
      return self.calculateLeftPadding(layoutDirection) + other.calculateLeftPadding(layoutDirection)
    }

    override fun calculateTopPadding(): Dp {
      return self.calculateTopPadding() + other.calculateTopPadding()
    }

    override fun calculateRightPadding(layoutDirection: LayoutDirection): Dp {
      return self.calculateRightPadding(layoutDirection) + other.calculateRightPadding(layoutDirection)
    }

    override fun calculateBottomPadding(): Dp {
      return self.calculateBottomPadding() + other.calculateBottomPadding()
    }
  }
}
```

**Comment by mattelacchiato** [comment]
I would question if this rule should be applied to file which does not contain any class at all. In earlier versions, this rule wasn't applied in this case and I would really like to keep it this way (for the already provided reasons here).

**Comment by PaulWoitaschek** [comment]
Yeah I'd prefer the rules to be split. Currently it's all or nothing but I'd like to apply some rules and others not.

**Comment by simjohan** [comment]
When can we expect a new tag including this fix?

**Comment by paul-dingemans** [comment]
> When can we expect a new tag including this fix?

I don't know. You can track the progress of what I planned to do in this [release](https://github.com/pinterest/ktlint/milestone/16)
